active/0000-keyword-arguments.md

```diff
@@ -0,0 +1,139 @@
+- Start Date: 2015-01-26
+- RFC PR #: (leave this empty)
+- Rust Issue #: (leave this empty)
+
+
+# Summary
+
+Add keyword arguments to Rust in a backwards-compatible way.
+This allows overloading that is safe in regards to type inference while being decided at compile time.
+
+# Motivation
+
+Allow another kind of argument in Rust. Current arguments distinguish themselves from other types by their order.
+If there is no semantic reason for a certain order, the argument order in functions is basically arbitrary.
+Consider a call like this:
+```rust
+window.addNewControl("Title", 20, 50, 100, 50, true);
+```
+
+First of all, the call tells nothing the user about what those values mean.
+Secondly, their order is arbitrary and must be memorized. What I propose is that this call would rather look like:
+```rust
+window.addNewControl(
+    title => "Title", 
+    xPosition => 20, 
+    yPosition => 50, 
+    width => 100, 
+    height => 50, 
+    drawingNow => true);
+```
+
+While you might argue that this is more verbose, this is already the standard in the JavaScript ecosystem.
+A lot of libraries in JavaScript actually have a convention with calling with associative arrays like:
+```JavaScript
+window.addNewControl({ xPosition: 20, yPosition: 50, width: 100, height: 5,
+                 drawingNow: true });
+```
+If this wasn't a convenient pattern, nobody would bother to do it.
```

Comment:
Overloading in Rust is IMHO absolutely unneded. We should use meaningful function names instead of that. Keyword arguments are different way of using object factory and IMHO there is no need for them as we can easily simulate them by using struct like @DanielKeep said. Only (and controversial) income from keyword arguments would be that we can reorder arguments in any way, i.e.

```rust
fn from_cartesian(x: i64, y: i64) -> Self { … }

Point::from_cartesian(y => 2, x => 4)
```

And that will save us from checking documentation for arguments order. I would oppose this if it will be tool to add function overloading as I consider it harmful.